core/vm/evm.go

```diff
@@ -186,9 +171,19 @@ type EVM struct {
 	callGasTemp uint64
 }
 
+type ChainConfig interface {
```

Comment:
Why is this being used instead of an actual `params.ChainConfig`?

Reply:
It is so the core/vm package would be decoupled from params package and can be either geth's original or subnet-evm/params.

Reply:
IMO to decouple the geth code from our code, a first important step is to avoid passing avalanche configurations to `core/vm`.

To ensure this, `.../core/vm` should not depend on the modified chain config type (`github.com/ava-labs/subnet-evm/params.ChainConfig`)

Later we can use the libevm/params.ChainConfig` or `go-ethereum/params.ChainConfig` type here, but that would require the `github.com/ava-labs/subnet-evm/params.ChainConfig` to be able to convert itself to one of these types, and I would like to do that refactor as a separate concern.